nordicsemi/dfu/dfu_transport_ant.py

```diff
@@ -0,0 +1,604 @@
+            if (mesg.msgid == antmessage.MESG_ADV_BURST_DATA_ID):
+                data = mesg.data[1:]
+
+        if is_first:
+            self.rx_data = data
+        else:
+            self.rx_data += data
+
+        if is_last:
+            self.__process_resp()
+            self.rx_data = None
+
+    def __process_resp(self):
+        (size, seq) = struct.unpack('<HB', bytearray(self.rx_data[:3]))
+
+        if seq == self.rx_seq:
+            logger.debug("Duplicate response received")
+            return
+
+        self.rx_seq = seq
+        self.resp_queue.put(self.rx_data[3:size])
+
+    def __process_evt(self, evt):
+        if (evt == antdefines.EVENT_CHANNEL_CLOSED):
+            raise NordicSemiException("Device connection lost")
+        elif (evt == antdefines.EVENT_TRANSFER_TX_COMPLETED):
+            self.tx_result = True
+        elif (evt == antdefines.EVENT_TRANSFER_TX_FAILED):
+            self.tx_result = False
+        elif (evt == antdefines.EVENT_TRANSFER_RX_FAILED):
+            self.rx_data = None
+
+    def __get_ant_mesg(self, timeout):
+        mesg = None
+        if self.ant_dev.wait_for_message(int(timeout * 1000)):
+            mesg = self.ant_dev.get_message()
+
+        if not mesg:
+            raise NordicSemiException('No message received from device')
+
+        return mesg
+
+
+class DfuTransportAnt(DfuTransport):
+    ANT_RST_TIMEOUT_MS          = 500
+    DEFAULT_PORT                = 0
+    DEFAULT_CMD_TIMEOUT         = 5.0   # Timeout on waiting for a response.
+    DEFAULT_SEARCH_TIMEOUT      = 10.0  # Timeout when searching for the device.
+    DEFAULT_PRN                 = 0
+    DEFAULT_DO_PING             = True
+    DEFAULT_DO_DEBUG            = False
+
+    OP_CODE = {
+        'CreateObject'          : 0x01,
+        'SetPRN'                : 0x02,
+        'CalcChecSum'           : 0x03,
+        'Execute'               : 0x04,
+        'ReadError'             : 0x05,
+        'ReadObject'            : 0x06,
+        'GetSerialMTU'          : 0x07,
+        'WriteObject'           : 0x08,
+        'Ping'                  : 0x09,
+        'Response'              : 0x60,
+    }
+
+    def __init__(self,
+                 ant_config=None,
+                 port=DEFAULT_PORT,
+                 timeout=DEFAULT_CMD_TIMEOUT,
+                 search_timeout=DEFAULT_SEARCH_TIMEOUT,
+                 prn=DEFAULT_PRN,
+                 debug=DEFAULT_DO_DEBUG):
+
+        super(DfuTransportAnt, self).__init__()
+        if ant_config is None:
+            ant_config = AntParams()
+        self.ant_config     = ant_config
+        self.port           = port
+        self.timeout        = timeout
+        self.search_timeout = search_timeout
+        self.prn            = prn
+        self.ping_id        = 0
+        self.dfu_adapter    = None
+        self.mtu            = 0
+        self.debug          = debug
+
+
+    def open(self):
+        super(DfuTransportAnt, self).open()
+        try:
```

Comment:
If the constructor ANTDevice fails to open a device it leads to a runtime error on line 357: `local variable 'ant_dev' referenced before assignment`
```suggestion
        ant_dev = None
        try:
```